src/ReactApp.mlx

```diff
@@ -0,0 +1,14 @@
+module App = struct
+  let make () =
+    ["Hello " ^ World.name ^ "!"; "This is React with OCaml and .mlx syntax!"]
+    |> List.map (fun greeting -> <h1> (React.string greeting) </h1>)
+    |> Array.of_list
+    |> React.array [@@react.component]
```

Comment:
Pretty sure you had this in mind already, but would be nice to have an API similar to the one jsoo-react [implemented](https://github.com/ml-in-barcelona/jsoo-react/blob/aa0c590469cf3bec0b148615b967bd0903586b3d/test/test_ml.ml#L159):

```ml
let%component make ~a ~b =
  React.Hooks.use_effect ~on:(a, b) (fun () -> incr count);
  div [||] []
```

Reply:
Yeah, I have this implemented in [react_of_ocaml](https://github.com/andreypopp/react_of_ocaml/blob/data/example/browser/example.ml#L10). For regular components, async components (for RSC) and for components exported as client components to server (in this case type annotations are required for arguments).

Or are you suggesting to implement this in reason-react ppx as well?

Reply:
No, this is what I meant. To be honest, I am not sure what's the place for reason-react anymore, after `mlx`. 😕